From: Petr Tesarik <ptesarik@suse.cz>
Subject: Allow reading/writing all memory through /dev/mem
Patch-Mainline: no
References: FATE#310031

With all pieces in place, I can now change the type of the physical
address pointer and get access to all memory.

Signed-off-by: Petr Tesarik <ptesarik@suse.cz>
Automatically created from "patches.suse/dev_mem-read-write-phys_addr_t.patch" by xen-port-patches.py

--- sle11sp2-2011-08-22.orig/drivers/xen/char/mem.c	2011-08-22 11:28:02.000000000 +0200
+++ sle11sp2-2011-08-22/drivers/xen/char/mem.c	2011-08-22 11:32:31.000000000 +0200
@@ -71,7 +71,7 @@ static inline int range_is_allowed(unsig
 static ssize_t read_mem(struct file *file, char __user *buf,
 			size_t count, loff_t *ppos)
 {
-	unsigned long p = *ppos;
+	phys_addr_t p = *ppos;
 	ssize_t read = 0, sz;
 	void __iomem *v;
 
@@ -120,7 +120,8 @@ static ssize_t read_mem(struct file *fil
 static ssize_t write_mem(struct file *file, const char __user *buf,
 			 size_t count, loff_t *ppos)
 {
-	unsigned long p = *ppos, ignored;
+	phys_addr_t p = *ppos;
+	unsigned long ignored;
 	ssize_t written = 0, sz;
 	void __iomem *v;
 
